_docs/protocols/2017-01-01-application-level.md

```diff
@@ -51,6 +51,40 @@ module.
 Every message type should have an instance of the `Message` typeclass.
 Please see [Time-Warp-NT guide](/protocols/time-warp-nt/#messaging).
 
+## Message names
+
+All messages are given custom names, since using full type names would be excessive. Each name is concatenation of one or two encoded `UnsignedVarInt`s.
+
+This table contains names for all used messages / message parts; they could also be found in
+[Pos.Communication.Message](https://github.com/input-output-hk/cardano-sl/blob/73d1b0a4281dd5453465304ed117b7127f82f79f/src/Pos/Communication/Message.hs) module.
+To distinguish from integers addition, concatenation is denoted here as `(++)`.
+
+| Message type | Name |
+|----------------------------|---------|--------------|
+| NOP | `0` |
+| SendProxySK | `2` |
+| ConfirmProxySK | `3` |
+| MsgGetHeaders | `4` |
+| MsgHeaders | `5` |
+| MsgGetBlocks | `6` |
+| MsgBlock | `7` |
+| InvMsg | `8` ++ `pMessageName tag` |
```

Comment:
Ops, `pMessageName` is introduced below. I'll move section `Message Name` so that `pMessageName` is in scope